tests/utils_tests/test_lazyobject.py

```diff
@@ -312,12 +312,26 @@ def test_repr(self):
         # __repr__ contains __repr__ of setup function and does not evaluate
         # the SimpleLazyObject
         self.assertRegex(repr(obj), '^<SimpleLazyObject:')
-        self.assertIs(obj._wrapped, empty)  # make sure evaluation hasn't been triggered
+        # Make sure evaluation hasn't been triggered
+        self.assertIs(obj._wrapped, empty)
 
         self.assertEqual(obj, 42)  # evaluate the lazy object
         self.assertIsInstance(obj._wrapped, int)
         self.assertEqual(repr(obj), '<SimpleLazyObject: 42>')
 
+    def test_add(self):
+        obj1 = self.lazy_wrap(1)
+
+        self.assertEqual(obj1 + 1, 2)
+
+    def test_radd(self):
+        obj1 = self.lazy_wrap(1)
+        obj2 = self.lazy_wrap(2)
+
+        self.assertEqual(obj2 + obj1, 3)
+        self.assertEqual(obj1 + obj2, 3)
```

Comment:
Are both of these going through the `__radd__` path? At a glance, I'd have _guessed_ that both are using `__add__` because the LHS is a `SimpleLazyObject` which now has that magic method? I could be wrong, it has been a _long_ time since I filed the ticket, so I may be misremembering.

Reply:
That is a nice question. I started by writing a test for `__add__` and had these assertions there. Not having a `__radd__` method gave the following error: `TypeError: unsupported operand type(s) for +: 'int' and 'SimpleLazyObject'` so I just moved them to the `__radd__` test so I assumed it went through the `__radd__` path.

The answer is that they are going through both paths. For example `obj1 + obj2` will evaluate to `obj1.__add__(obj2)`. `__add__` first evaluates the object via `new_method_proxy` and calls the `add` method from the operator module. This method basically just adds the two inputs so then the LHS is a number (obj1 has been evaluated) but the RHS is a `SimpleLazyObject ` and this will call `__radd__`.

I guess looking at it now, having both assertions is redundant. Maybe this needs to be its own test like `test_add_and_radd` with a docstring explanation or be removed completely and just have the next assertion which is `self.assertEqual(1 + obj1, 2)`.

Reply:
I'll leave the details of splitting and docstring commenting for others to assist with deciding on, but thanks so much for the detailed explanation; there's enough magic going on between the SLO and dunder methods that just reading the diff wasn't illuminating enough.